src/NLog.Layouts.GelfLayout/GelfConverter.cs

```diff
@@ -97,7 +103,7 @@ public void ConvertToGelfMessage(JsonWriter jsonWriter, LogEventInfo logEventInf
             ICollection<string> mdlcKeys = null;
             if (converterOptions.IncludeScopeProperties)
             {
-                mdlcKeys = MappedDiagnosticsLogicalContext.GetNames();
+                mdlcKeys = ScopeContext.GetAllProperties().Select(x => x.Key).ToList();

```

Comment:
Maybe check the result from `ScopeContext.GetAllProperties()` is empty, before doing `Select` and `ToList`